Riot/Modules/Pills/PillsFormatter.swift

```diff
@@ -80,25 +84,27 @@ class PillsFormatter: NSObject {
                                        mode: PillsReplacementTextMode = .displayname) -> String {
         let newAttr = NSMutableAttributedString(attributedString: attributedString)
         newAttr.vc_enumerateAttribute(.attachment) { (attachment: PillTextAttachment, range: NSRange, _) in
-            if let displayText = attachment.data?.displayText,
-               let userId = attachment.data?.matrixItemId,
-               let permalink = MXTools.permalinkToUser(withUserId: userId) {
-                let pillString: String
-                switch mode {
-                case .displayname:
-                    pillString = displayText
-                case .identifier:
-                    pillString = userId
-                case .markdown:
-                    pillString = "[\(displayText)](\(permalink))"
-                }
-                newAttr.replaceCharacters(in: range, with: pillString)
+            guard let data = attachment.data else {
+                return
             }
+
+            var pillString: String
```

Comment:
```suggestion
            let pillString: String
```

Reply:
> Can you take a look at this failing test: `- (void)testLinkWithRoomAliasLink` ? Seems related to this change.
> 
> Otherwise LGTM

Ok, I have updated the test because now we want to have a link for all permalinks, so we can render pills.